### PR TITLE
pscan: migrate core API implementor

### PR DESCRIPTION
**Pull request by thc202**
Add base add-on and migrate core API implementor.

Part of zaproxy/zaproxy#7959.

### CODE REVIEW

File: addOns/pscan/gradle.properties [hunk 1]
```diff
@@ -0,0 +1,4 @@
+version=0.0.1
+release=false
+zap.maven.publish=true
+zap.maven.pom.inceptionyear=2024
```
Comment (kingthorin) [inline addOns/pscan/gradle.properties:4]: Should we be adding this on others?

File: addOns/pscan/pscan.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import org.zaproxy.gradle.addon.AddOnStatus
+
+description = "Provides core passive scanning capabilities."
+
+zapAddOn {
+    addOnName.set("Passive Scanner")
+    addOnStatus.set(AddOnStatus.ALPHA)
```
Comment (kingthorin) [inline addOns/pscan/pscan.gradle.kts:7]: Release?

Comment (kingthorin) [reply addOns/pscan/pscan.gradle.kts:7]: Or this is just "for now"?

File: addOns/pscan/src/main/java/org/zaproxy/addon/pscan/PassiveScanApi.java
Comment (kingthorin) [inline addOns/pscan/src/main/java/org/zaproxy/addon/pscan/PassiveScanApi.java:1]: Is this pretty much a literal copy or do I need to read through?

File: addOns/pscan/pscan.gradle.kts (hunk 2 above)
Comment (thc202) [reply addOns/pscan/pscan.gradle.kts:7]: The latter.

File: addOns/pscan/gradle.properties (hunk 1 above)
Comment (thc202) [reply addOns/pscan/gradle.properties:4]: This is only needed in add-ons that are expected to be released to Maven Central, this one should be for other add-ons to depend on it in time.

**Review by kingthorin — APPROVED** [review]

**Review by psiinon — APPROVED** [review]

**Pull request merged**